community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/RelationshipTypeCheckerStep.java

```diff
@@ -42,13 +40,13 @@
  */
 public class RelationshipTypeCheckerStep extends ProcessorStep<Batch<InputRelationship,RelationshipRecord>>
 {
-    private static final Comparator<Map.Entry<Object,AtomicLong>> SORT_BY_COUNT_DESC =
-            (e1,e2) -> Long.compare( e2.getValue().get(), e1.getValue().get() );
-    private static final Comparator<Map.Entry<Object,AtomicLong>> SORT_BY_ID_DESC =
+    private static final Comparator<Map.Entry<Object,MutableLong>> SORT_BY_COUNT_DESC =
+            (e1,e2) -> Long.compare( e2.getValue().value, e1.getValue().value );
+    private static final Comparator<Map.Entry<Object,MutableLong>> SORT_BY_ID_DESC =
             (e1,e2) -> Integer.compare( (Integer)e2.getKey(), (Integer)e1.getKey() );
-    private final ConcurrentMap<Object,AtomicLong> allTypes = new ConcurrentHashMap<>();
+    private final Map<Thread,Map<Object,MutableLong>> typeCheckers = new ConcurrentHashMap<>();
```

Comment:
We don't know before-hand and it may change throughout the import. Items in that global map will change only when changing number of workers. I.e. this ain't an issue.